### PR TITLE
cleanup: Ensure all jsx and js files pass eslint

### PR DESCRIPTION
**Pull request by traviskaufman**
* Add plugins to eslintrc
* Modify `npm run lint` cmd to include jsx files
* Fix all lint errors in source files

### CODE REVIEW

**Comment by traviskaufman** [comment]
This was a good way to get myself familiarized with the codebase 😄 I'm happy to split this up into smaller PRs if that's preferred.

File: .eslintrc [hunk 1]
```diff
@@ -7,5 +7,10 @@
     "no-alert": ["off"],
     "jsx-quotes": ["error", "prefer-single"],
     "no-confusing-arrow": ["off"]
-  }
-}
\ No newline at end of file
+  },
+  "plugins": [
```
Comment (saikat) [inline .eslintrc]: Does the base airbnb eslint not include these plugins?

**Comment by saikat** [comment]
Yeah this is often a way I learn new codebases too.  Thanks!  Left one small note, but looks good to merge otherwise.

**Comment by traviskaufman** [comment]
@saikat comments addressed! 💎

**Comment by saikat** [comment]
Thanks!

**Pull request merged**